File: bluetooth/src/main/java/com/ankerboxmanager/bluetooth/IBluetoothLe.java
//package com.ankerboxmanager.bluetooth;
//
//import java.util.List;
//
//import com.ankerboxmanager.bluetooth.bean.AnkerBoxBluetoothGattCharacteristic;
//import com.ankerboxmanager.bluetooth.bean.AnkerBoxBluetoothLeService;
//
//import android.content.Context;
//
///**
// * Created on 21/03/2018.
// *
// * @author lx
// */
//
//public interface IBluetoothLe extends IBluetooth {
//
//    interface StateChangedListener {
//        void onChanged(String deviceId, int state);
//    }
//
//    interface CharacteristicValueChangedListener {
//        void onChanged(String deviceId, String serviceId, String characteristicId, String value);
//    }
//
//    boolean isSupportLe();
//
////    void startLeDevicesDiscovery(String[] serviceUUIDs, AsyncReturnCallback returnCallback);
//    //
//    //    void stopLeDevicesDiscovery(AsyncReturnCallback returnCallback);
//
//    void connectGatt(Context context, String deviceId, AsyncReturnCallback returnCallback);
//
//    boolean disconnectGatt(Context context, String deviceId);
//
//    boolean addLeStateChangedListener(StateChangedListener listener);
//
//    boolean removeLeStateChangedListener(StateChangedListener listener);
//
//    List<AnkerBoxBluetoothLeService> getServices(Context context, String deviceId);
//
//    List<AnkerBoxBluetoothGattCharacteristic> getDeviceCharacteristics(Context context, String deviceId,
//                                                                       String serviceId);
//
//    boolean readCharacteristicValue(Context context, String deviceId, String serviceId, String characteristicId);
//
//    boolean writeCharacteristicValue(Context context, String deviceId, String serviceId, String characteristicId,
//                                     String value);
//
//    boolean notifyCharacteristicValueChange(Context context, String deviceId, String serviceId, String characteristicId,
//                                            int state);
//
//    boolean addCharacteristicValueChangedListener(CharacteristicValueChangedListener listener);
//
//    boolean removeCharacteristicValueChangedListener(CharacteristicValueChangedListener listener);
//
//}

File: bluetooth/src/main/java/com/ankerboxmanager/bluetooth/IBluetoothLe.kt
package com.ankerboxmanager.bluetooth

import java.util.UUID

import com.ankerboxmanager.bluetooth.bean.AnkerBoxBluetoothGattCharacteristic
import com.ankerboxmanager.bluetooth.bean.AnkerBoxBluetoothLeService

import android.content.Context

/**
 * Created on 21/03/2018.
 *
 * @author lx
 */

interface IBluetoothLe : IBluetooth {

    val isSupportLe: Boolean

    interface StateChangedListener {
        fun onChanged(deviceId: String, state: Int)
    }

    interface CharacteristicValueChangedListener {
        fun onChanged(deviceId: String, serviceId: UUID, characteristicId: UUID, value: ByteArray)
    }

    fun createGattConnection(context: Context, deviceId: String, returnCallback: IBluetooth.AsyncReturnCallback?)

    fun closeLeConnection(context: Context, deviceId: String): Boolean

    fun addLeStateChangedListener(listener: StateChangedListener): Boolean

    fun removeLeStateChangedListener(listener: StateChangedListener): Boolean

    fun getServices(context: Context, deviceId: String): List<AnkerBoxBluetoothLeService>?

    fun getDeviceCharacteristics(context: Context, deviceId: String,
                                 serviceId: UUID): List<AnkerBoxBluetoothGattCharacteristic>?

    fun readCharacteristicValue(context: Context, deviceId: String, serviceId: UUID, characteristicId: UUID): Boolean

    fun writeCharacteristicValue(context: Context, deviceId: String, serviceId: UUID, characteristicId: UUID,
                                 value: String): Boolean

    fun notifyCharacteristicValueChange(context: Context, deviceId: String, serviceId: UUID, characteristicId: UUID,
                                        state: Int): Boolean

    fun addCharacteristicValueChangedListener(listener: CharacteristicValueChangedListener): Boolean

    fun removeCharacteristicValueChangedListener(listener: CharacteristicValueChangedListener): Boolean
}
